miner/src/main/java/fr/raksrinana/channelpointsminer/miner/database/MariaDBDatabase.java

```diff
@@ -39,6 +42,44 @@ public void initDatabase() throws SQLException{
 							INDEX `EventDateIdx`(`EventDate`),
 							INDEX `EventTypeIdx`(`Type`)
 						)
+						ENGINE=InnoDB DEFAULT CHARSET=utf8;""",
+				"""
+						CREATE TABLE IF NOT EXISTS `ResolvedPrediction` (
+							`EventID` VARCHAR(36) NOT NULL PRIMARY KEY,
+							`ChannelID` VARCHAR(32) NOT NULL REFERENCES `Channel`(`ID`),
+							`Title` VARCHAR(64) CHARACTER SET utf8mb4 NOT NULL,
+							`EventCreated` DATETIME NOT NULL,
+							`EventEnded` DATETIME NULL,
+							`Canceled` BOOLEAN NOT NULL,
+							`Outcome` VARCHAR(32) CHARACTER SET utf8mb4 NULL,
+							`Badge` VARCHAR(32) NULL,
+							`ReturnRatioForWin` DOUBLE NULL,
+							INDEX `ChannelIDIdx`(`ChannelID`)
+						)
+						ENGINE=InnoDB DEFAULT CHARSET=utf8;""",
+				"""
+						CREATE TABLE IF NOT EXISTS `PredictionUser` (
+							`ID` INT NOT NULL PRIMARY KEY AUTO_INCREMENT,
+							`Username` VARCHAR(128) NOT NULL,
+							`PredictionCnt` SMALLINT UNSIGNED NOT NULL DEFAULT 0,
+							`WinCnt` SMALLINT UNSIGNED NOT NULL DEFAULT 0,
+							`WinRate` DECIMAL(8,7) NOT NULL DEFAULT 0,
+							`ReturnOnInvestment` DOUBLE NOT NULL DEFAULT 0,
+							UNIQUE (`Username`),
+							INDEX `UsernameIdx`(`Username`)
+						)
+						ENGINE=InnoDB DEFAULT CHARSET=utf8;""",
+				"""
+						CREATE TABLE IF NOT EXISTS `UserPrediction` (
+							 `ChannelID` VARCHAR(32) NOT NULL REFERENCES `Channel`(`ID`),
+							 `UserID` INT NOT NULL REFERENCES `PredictionUser`(`ID`),
+							 `ResolvedPredictionID` VARCHAR(36) NOT NULL DEFAULT '',
+							 `Badge` VARCHAR(32) NOT NULL,
+							 PRIMARY KEY (`ChannelID`, `UserID`, `ResolvedPredictionID`),
+							 INDEX `ChannelIDIdx`(`ChannelID`),
+							 INDEX `UserIDIdx`(`UserID`),
+							 INDEX `ResolvedPredictionIDIdx`(`ResolvedPredictionID`)
+						)
 						ENGINE=InnoDB DEFAULT CHARSET=utf8;""");
```

Comment:
Instead of using `VARCHAR CHARACTER SET utf8mb4` I think we can instead use `ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;` for all the tables. Like that we're a bit more failproof towards what twitch strings may contains.

Reply:
Changed it for Mariadb. The SQLite can already handle these characters.